### DISCUSSION TITLE
Long build time for medium to large e-commerce site

### DISCUSSION THREAD

**Discussion by tiwac100** [vercel/vercel#4191] [General] [answered]
Hello,

we are running a Next.js shop on ZEIT with ~16k products and the current build with only ~2.5k products enabled takes ca. 50 minutes.

What kind of optimizations could be done to improve build times?

Some ideas that come to my mind:
1. Providing **real** server side fallbacks for lazy building (the current fallback approach for static pages is unusable for SEO purposes since only the loading page is sent to the browser - a usable approach would be to wait on the server side until static props are generated and send a full response)
2. Incremental builds (well, technically the same as the on-demand build without the current fallback implementation)
3. Parallel processing

**Comment by coetry** [comment, marked as answer]
> the current fallback approach for static pages is unusable for SEO purposes since only the loading page is sent to the browser

This is 100% a myth as far as Google is concerned. Googlebot does a two-pass crawl for SPAs with client side javascript and the second pass crawls the generated HTML.

Generating 2500 SSG pages is definitely a lot and I would recommend only SSGing a handset of popular pages and using `fallback: true` for the remainder. Note that once a page is dynamically generated, it is treated as static HTML from that point on.
